### PR TITLE
Use apm-server latest for macro benchmarks

### LINKED ISSUE #10315: Build current branch in APM macro Benchmarks
## Description

The APM macro Benchmarks currently use the last committed Elastic Agent docker image for the APM Server's daily benchmarks https://github.com/elastic/apm-server/blob/52b02b7323b0026b08f279c680c59a5a2ba35bfe/docker-compose.yml#L53 https://github.com/elastic/apm-server/blob/52b02b7323b0026b08f279c680c59a5a2ba35bfe/testing/benchmark/Makefile#L119-L122.

Making the macro benchmarks overly reliant on the stack update workflow and assumes that unified stack builds are mostly successful, which is not always the case. This has caused the APM Server latest changes to not be benchmarked for several days, which is less than ideal.

Additionally, if we don't schedule the benchmarks to be run a few hours after the stack bump happens, we may have an unnecessarily long feedback cycle.

## Proposed path forward

To reduce dependencies, pains, and have a faster feedback cycle on the changes that are committed to the APM Server, I propose that we build the local branch changes, package it in a docker image, and use that docker image for the APM Server benchmarks. We should keep make use of the committed stack hash version since that assures us that Stack components are healthy according to our system tests.

We already have packaging and docker image building in place in [`testing/cloud`](https://github.com/elastic/apm-server/tree/main/testing/cloud#building-and-deploying-a-custom-elastic-agent-image), we only need to polish and adapt it so it can be used in `testing/benchmark`.

Once we've closed this issue, it will also enable us to run the automated benchmarks on demand in PRs (with adequate protections).

### PR DESCRIPTION
**Pull request by endorama**
## Motivation/summary

The APM macro Benchmarks currently use the last committed Elastic Agent docker image for the APM Server's daily benchmarks. This introduce a dependency on all other components to be built correctly to use the latest `apm-server` code: in case a dependant build fails, the Elastic Agent docker image is not updated. The result of this behaviour is that macro benchmarks may target a older-than-main version of `apm-server`, delaying the feedback loop on recent code changes.

This PR aims at removing this dependency by always building an Elastic Agent docker container that contains the latest `apm-server` code from the `main` branch of this repository. As macro benchmarks target `apm-server` performances, we don't care too much about the version of the Elastic Agent (as long as is reasonably new - a couple of weeks old is ok) but we care to run benchmarks against the most recent version of `apm-server`.

This PR updates the `Make` targets currently used to override docker image tags to build custom Kibana and Elastic Agent images with the latest `apm-server` code.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## Related issues

Closes #10315

### CODE REVIEW

**Comment by dmathieu** [comment]
Should `testing/benchmark/README.md` be updated as well?

**Comment by endorama** [comment]
Latests macro benchmark test run completed successfully https://apm-ci.elastic.co/job/apm-server/job/benchmarks/view/change-requests/job/PR-10816/12/

**Comment by endorama** [comment]
Run macro benchmark build [#13](https://apm-ci.elastic.co/job/apm-server/job/benchmarks/view/change-requests/job/PR-10816/13/)

File: testing/benchmark/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+docker_image.auto.tfvars
```
Comment (kruskall) [inline testing/benchmark/.gitignore:1]: suggestion: some of the files in this folder are ignored in the main `.gitignore`. Should we move those lines here for consistency ?

**Review by kruskall — COMMENTED** [review]
LGTM! Just a minor suggestion

**Review by kruskall — APPROVED** [review]

File: testing/benchmark/.gitignore (hunk 1 above)
Comment (endorama) [reply testing/benchmark/.gitignore:1]: Thanks, moved!

**Comment by endorama** [comment]
Rebased as I needed to sign all commits

**Pull request merged**